lib/nodes/Decorator.ts

```diff
@@ -27,7 +28,7 @@ export class Decorator<S extends BlueshellState, E> extends Composite<S, E> {
 	 * @param state
 	 * @param event
 	 */
-	onEvent(state: S, event: E): string|Promise<string> {
+	handleChild(state: S, event: E): ResultCode {
```

Comment:
this is a _very_ significant change, isn't it? 🤔

Reply:
now we have the latched/storage.running functionality of Composite.onEvent

Reply:
It might be a good idea to leave `onEvent` here in this PR, even if that's wrong, and then handle that as a separate issue. Or maybe we should just go with this and fix it quickly if it breaks any existing decorators 😈

Reply:
Decorators aren't latched, so I think this is a pretty safe change. If it would make you feel better, we can explicitly set latched = false in Decorator.ts

Reply:
Good point, this is safe then 👍